d2-schemas/src/main/pegasus/com/linkedin/d2/D2Uri.pdl

```diff
@@ -19,4 +19,44 @@ record D2Uri {
    * partitionId key to weight
    */
   partitionDescription: map[string, double]
+
+  /**
+   * URI for this machine with the hostname replaced with an ipv4. Only set if an ipv4 is provided and isIpv4InSan is true.
+   */
+  ipv4VariantURI: optional string
+
+  /**
+   * URI for this machine with the hostname replaced with an ipv6. Only set if an ipv6 is provided and isIpv6InSan is true
+   */
+  ipv6VariantURI: optional string
+
+  /**
+   * The time this URI was last modified in milliseconds since the epoch
+   */
+  modifiedTime: optional long
+
+  /**
+   * The hostname of the machine making this announcement
+   */
+  hostname: optional string
+
+  /**
+   * The ipv4 of the machine making this announcement
+   */
+  ipv4Address: optional string
+
+  /**
+   * The ipv6 of the machine making this announcement
+   */
+  ipv6Address: optional string
+
+  /**
+   * Whether the ipv4 provided is also in the app's TLS cert
+   */
+  isIpv4InSan: optional boolean
+
+  /**
+   * Whether the ipv6 provided is also in the app's TLS cert
+   */
+  isIpv6InSan: optional boolean
```

Comment:
should have app specific properties too, some apps set custom values in it. It's also good for everyone to see what's set.

Reply:
You mean the `UriSpecificProperties`? We can look into it, though it may be tough to translate because the type is `Struct` which is way more flexible than any of the types we're including so far

Reply:
Looking closer into this, I see that a `Struct` can be arbitrarily nested... for simplicity and also considering nobody has yet requested this data, I would suggest we defer trying to add the `UriSpecificProperties` future work